datafusion/physical-plan/src/aggregates/group_values/multi_group_by/bytes_view.rs

```diff
@@ -226,14 +249,26 @@ impl<B: ByteViewType> ByteViewGroupValueBuilder<B> {
         let exist_null = self.nulls.is_null(lhs_row);
         let input_null = array.is_null(rhs_row);
         if let Some(result) = nulls_equal_to(exist_null, input_null) {
-            return result;
+            result
+        } else {
+            self.do_equal_to_inner_values_only(lhs_row, array, rhs_row)
```

Comment:
perhaps this could be optimized as well (in the outer loop) for empty data buffers

Reply:
Let's move this conversation to https://github.com/apache/datafusion/pull/19364